### PR TITLE
feat: browser refresh

### PR DESCRIPTION
**Pull request by tsaleksandrova**
todo: fix tests

### CODE REVIEW

File: src/uiveri5.js [hunk 1]
```diff
@@ -318,6 +318,25 @@ function run(config) {
         });
       };
 
+      browser.get = function (sUrl, iTimeout) {
```
Comment (maximnaidenov) [inline src/uiveri5.js]: what about the authentication ?

Comment (maximnaidenov) [inline src/uiveri5.js]: Overriding the browser.get is a change in our current "strategy" where we have browser.testrunner.navigate API, modeled according to browser.driver.navigate API.
I am generally fine but my concern is how to fit the declarative authentication and how to cope with the implicit expectation that every test should start with browser.get

File: src/scripts/clientsidescripts.js [hunk 2]
```diff
@@ -267,6 +267,14 @@ var mFunctions = {
       throw new Error('Log collection is not set up! Call "startLogCollection" before "stopLogsCollection"');
     }
     return window.uiveri5._BrowserLogCollector.stop();
+  },
+
+  setLocation: function setLocation (mScriptParams, fnCallback) {
+    var sCurrentHash = window.location.hash.substring(2);
+    if (sCurrentHash !== mScriptParams.hash) {
```
[the remark this answers is not in the document]
Comment (tsaleksandrova) [reply src/scripts/clientsidescripts.js]: will leave this out for now, as discussed

File: src/uiveri5.js (hunk 1 above)
Comment (tsaleksandrova) [reply src/uiveri5.js]: it's not always needed to authenticate; we could have another method for programmatic authentication

File: src/uiveri5.js [hunk 3]
```diff
@@ -322,19 +328,22 @@ function run(config) {
...
+        var authenticator =  moduleLoader.loadNamedModule(vAuth, [statisticCollector]);
+        // open page and login
+        browser.controlFlow().execute(function () {
+          logger.info('Opening: ' + sUrl);
         });
+        authenticator.get(sUrl);
+
+        // load waitForUI5 logic on client and
+        // ensure app is fully loaded before starting the interactions
+        return browser.loadUI5Dependencies();
```
Comment (maximnaidenov) [inline src/uiveri5.js]: where is the overloading of browser.refresh() and browser.reload(). I see the tests but no impl so wondering how this is working ?

File: src/uiveri5.js (hunk 1 above)
Comment (maximnaidenov) [reply src/uiveri5.js]: 1. no need for an explicit iTimeout setting, the global settings will apply.
2. Lets have a single .get method with optional auth param that defaults to plain. It should call navigate.to as we should keep it for backward compatibility

**Comment by maximnaidenov** [comment]
Please also add some docs - maybe a new browser.md file that shortly goes over browser management scenarios. Also please remove the mentioning of navigate.to from authentication.md and please refer to the new md.

**Comment by maximnaidenov** [comment]
Also, please rename the commit/pr to feat: support browser refresh

File: src/uiveri5.js (hunk 3 above)
Comment (tsaleksandrova) [reply src/uiveri5.js]: browser.refresh: gets the current url and calls browser.get with this url

Comment (maximnaidenov) [reply src/uiveri5.js]: ok but please mention in docs so the users know we support them

File: src/uiveri5.js (hunk 1 above)
Comment (tsaleksandrova) [reply src/uiveri5.js]: the auth param is supposed to contain the credentials. if a default strategy is used, or even if auth is just "enabled", credentials could be read from the config file but that would be very confusing to the user imo
navigate.to has some logging specific to the first page load. do we need browser.refresh/sleep  and version stamps in browser.get as well?

Comment (maximnaidenov) [reply src/uiveri5.js]: the auth param should contain the auth configuration that may include the credentials. Exactly the same as navigate.to with the difference that empty config = plain

Comment (maximnaidenov) [reply src/uiveri5.js]: I suggest that we directly call navigate.to with a default value if auth is missing

**Comment by tsaleksandrova** [comment]
if we want to use the initial page load parameters in every call to get, then i'm fine with navigate.to

**Pull request merged**